<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jing Gu" />

<meta name="date" content="2024-06-07" />

<title>u19_h2g_enrichment_analysis</title>

<script src="site_libs/header-attrs-2.26/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.33/datatables.js"></script>
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">lung_lymph_scMultiomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/j3gu/lung_lymph_scMultiomics">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">u19_h2g_enrichment_analysis</h1>
<h4 class="author">Jing Gu</h4>
<h4 class="date">2024-06-07</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-06-07
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>lung_lymph_scMultiomics/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221229code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221229)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221229code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221229)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomj3gulunglymphscMultiomicstreeede034029a4b012d85c49568c901350db96ca997targetblankede0340a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/tree/ede034029a4b012d85c49568c901350db96ca997" target="_blank">ede0340</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomj3gulunglymphscMultiomicstreeede034029a4b012d85c49568c901350db96ca997targetblankede0340a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/tree/ede034029a4b012d85c49568c901350db96ca997" target="_blank">ede0340</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    analysis/.RData
    Ignored:    analysis/.Rhistory

Untracked files:
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/test_GO_enrichment.ipynb
    Untracked:  analysis/u19_atac_fastTopics.Rmd
    Untracked:  analysis/u19_regulon_enrichment.Rmd
    Untracked:  data/TF_target_sizes_GRN.txt
    Untracked:  data/U19_T_cell_peaks_metadata.RDS
    Untracked:  data/Wang_2020_T_cell_peaks_metadata.RDS
    Untracked:  data/lung_GRN_CD4_T_edges.txt
    Untracked:  data/lung_GRN_CD8_T_edges.txt
    Untracked:  data/lung_GRN_Th17_edges.txt
    Untracked:  data/lung_GRN_Treg_edges.txt
    Untracked:  output/annotation_reference.txt
    Untracked:  output/fastTopics
    Untracked:  output/homer/
    Untracked:  output/ldsc_enrichment
    Untracked:  output/lung_immune_atac_peaks_high_ePIPs.RDS
    Untracked:  output/positions.bed

Unstaged changes:
    Modified:   analysis/cross_tissue_DE_u19_fastTopics.Rmd
    Modified:   analysis/identify_regulatory_programs_u19_GRN.Rmd
    Modified:   analysis/rank_TFs_from_pairwise_comparison.ipynb
    Deleted:    code/run_fastTopic.R
    Deleted:    lung_immune_fine_mapping.Rproj

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/u19_h2g_enrichment.Rmd</code>)
and HTML (<code>docs/u19_h2g_enrichment.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/ede034029a4b012d85c49568c901350db96ca997/analysis/u19_h2g_enrichment.Rmd" target="_blank">ede0340</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-06-07
</td>
<td>
added tissue-group analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/lung_lymph_scMultiomics/ca85982484a362036196c17bf218d91ecaea4872/docs/u19_h2g_enrichment.html" target="_blank">ca85982</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-06-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/32fec27fd554efa4897144c8fa33cd58a0b8f108/analysis/u19_h2g_enrichment.Rmd" target="_blank">32fec27</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-06-05
</td>
<td>
update h2g analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/lung_lymph_scMultiomics/71a11fc1d56cbc833a77f52979b2db80daabb2cb/docs/u19_h2g_enrichment.html" target="_blank">71a11fc</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-06-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/8e7a5acb456e8a39bdba33f7ddee763b03006d4c/analysis/u19_h2g_enrichment.Rmd" target="_blank">8e7a5ac</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-06-05
</td>
<td>
h2g analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="outlines" class="section level1">
<h1>Outlines</h1>
<ol style="list-style-type: decimal">
<li>Evaluate how much additional information we can get from our
multi-omics dataset</li>
</ol>
<ul>
<li>characterize and compare our identified open chromatin peaks with
Wang et al.</li>
<li>estimate heritability enrichment for annotations derived from our
chromatin data across many traits
<ul>
<li>cell-type based annotations</li>
<li>topic-based annotations</li>
</ul></li>
</ul>
<ol start="2" style="list-style-type: decimal">
<li>Identify relevant cell-types and tissues for Asthma</li>
</ol>
<ul>
<li>find peaks in our dataset that overlap with enhancers with high
ePIPs based on GWAS and other single-cell ATAC-seq data and then get the
corresponding cell-type and tissue information</li>
<li>compute ePIP using our own data set and then identify relevant
cell-types or tissues
<ul>
<li>Compute ePIPs by summing up PIPs of causal variants that overlap
with each enhancer</li>
</ul></li>
</ul>
</div>
<div
id="evaluate-the-heritability-contribution-of-our-multi-omics-dataset"
class="section level1">
<h1>Evaluate the heritability contribution of our multi-omics
dataset</h1>
<div
id="characterize-and-compare-peaks-from-t-cells-between-two-dataset"
class="section level2">
<h2>Characterize and compare peaks from T cells between two dataset</h2>
<p>A barplot of summarizing peak counts by types show Wang et
al. identified more peaks in distal and exonic regions but fewer ones in
promoters compared to our dataset. The color indicates whether each
query peak in U19 overlaps with the one in Wang et al. We see more than
50% peaks in U19 overlapped with ones in Wang et al., while these
overlapped peaks take up less than 50% of peaks in Wang et al.
<img src="figure/u19_h2g_enrichment.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/71a11fc1d56cbc833a77f52979b2db80daabb2cb/docs/figure/u19_h2g_enrichment.Rmd/unnamed-chunk-2-1.png" target="_blank">71a11fc</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-06-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="identify-relevant-tissue-or-cell-type-for-immune-diseases"
class="section level2">
<h2>Identify relevant tissue or cell-type for immune diseases</h2>
<p>METHOD: Test heritability enrichment through S-LDSC</p>
<p>The union set of peaks from U19 atac-seq data were first lifted over
to hg19. The following table shows the change in genome builds only made
1% of peaks have inconsistent width.</p>
<pre><code>                                nPeak_difference nPeak_hg19
lungs_Type_17_helper_T_cells                   0       1208
lungs_Tem_Trm_cytotoxic_T_cells              -58      10539
lungs_Tcm_Naive_helper_T_cells               -35      13234
lungs_Regulatory_T_cells                     -44      12322
lungs_Naive_B_cells                          -85      18411
spleens_Regulatory_T_cells                   -16       8348
                                percentPeak_unequal_width
lungs_Type_17_helper_T_cells                        0.017
lungs_Tem_Trm_cytotoxic_T_cells                     0.011
lungs_Tcm_Naive_helper_T_cells                      0.013
lungs_Regulatory_T_cells                            0.010
lungs_Naive_B_cells                                 0.012
spleens_Regulatory_T_cells                          0.014</code></pre>
<div id="test-enrichment-for-individual-cell-type-of-each-tissue"
class="section level3">
<h3>Test enrichment for individual cell-type of each tissue</h3>
<ul>
<li>GWAS: a broad range of triats</li>
<li>Baseline annotation: 53 annotations from Finucane et al. 2015 Nat
Genet (Baseline v1.2)</li>
</ul>
<p>Overall, we observed immune cells in both tissues are significantly
enriched for genetic risks of immune diseases but not those of other
traits. T cells from both tissues show significant enrichment for risk
variants of immune diseases, while only lung B cells show
enrichment.</p>
<p><img src="figure/u19_h2g_enrichment.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/71a11fc1d56cbc833a77f52979b2db80daabb2cb/docs/figure/u19_h2g_enrichment.Rmd/unnamed-chunk-5-1.png" target="_blank">71a11fc</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-06-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/u19_h2g_enrichment.Rmd/unnamed-chunk-5-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-2">
Past versions of unnamed-chunk-5-2.png
</button>
</p>
<div id="fig-unnamed-chunk-5-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/71a11fc1d56cbc833a77f52979b2db80daabb2cb/docs/figure/u19_h2g_enrichment.Rmd/unnamed-chunk-5-2.png" target="_blank">71a11fc</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-06-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/u19_h2g_enrichment.Rmd/unnamed-chunk-5-3.png" width="672" style="display: block; margin: auto;" /></p>
<div class="datatables html-widget html-fill-item" id="htmlwidget-e3dd2f708ffc37673119" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-e3dd2f708ffc37673119">{"x":{"filter":"none","vertical":false,"data":[["9880","9881","9882","9883","9884","9885","9886","9887","9888","9889","9890","9891","9892","9893","9894","9895","9896","9897","9898","9899","98100","98101","98102","98103","98104","98105","98106","98107","98108","98109","98110","98111","98112","98113","98114","98115","98116","98117","98118","98119","98120","98121","98122","98123","98124","98125","98126","98127"],["allergy","allergy","allergy","allergy","allergy","allergy","allergy","allergy","allergy","allergy","allergy","allergy","allergy","allergy","allergy","allergy","aoa_ukb","aoa_ukb","aoa_ukb","aoa_ukb","aoa_ukb","aoa_ukb","aoa_ukb","aoa_ukb","aoa_ukb","aoa_ukb","aoa_ukb","aoa_ukb","aoa_ukb","aoa_ukb","aoa_ukb","aoa_ukb","coa_ukb","coa_ukb","coa_ukb","coa_ukb","coa_ukb","coa_ukb","coa_ukb","coa_ukb","coa_ukb","coa_ukb","coa_ukb","coa_ukb","coa_ukb","coa_ukb","coa_ukb","coa_ukb"],["lungs_CD16_neg_NK","lungs_CD16_pos_NK","lungs_Memory_B","lungs_Naive_B","lungs_CD4_T","lungs_CD8_T","lungs_Th17","lungs_Treg","spleens_CD16_neg_NK","spleens_CD16_pos_NK","spleens_Memory_B","spleens_Naive_B","spleens_CD4_T","spleens_CD8_T","spleens_Th17","spleens_Treg","lungs_CD16_neg_NK","lungs_CD16_pos_NK","lungs_Memory_B","lungs_Naive_B","lungs_CD4_T","lungs_CD8_T","lungs_Th17","lungs_Treg","spleens_CD16_neg_NK","spleens_CD16_pos_NK","spleens_Memory_B","spleens_Naive_B","spleens_CD4_T","spleens_CD8_T","spleens_Th17","spleens_Treg","lungs_CD16_neg_NK","lungs_CD16_pos_NK","lungs_Memory_B","lungs_Naive_B","lungs_CD4_T","lungs_CD8_T","lungs_Th17","lungs_Treg","spleens_CD16_neg_NK","spleens_CD16_pos_NK","spleens_Memory_B","spleens_Naive_B","spleens_CD4_T","spleens_CD8_T","spleens_Th17","spleens_Treg"],["0.0272%","0.2552%","0.1338%","0.3141%","0.2172%","0.1755%","0.0196%","0.2019%","0.0303%","0.0184%","0.0552%","0.0453%","0.0408%","0.0096%","0.0248%","0.1361%","0.0272%","0.2552%","0.1338%","0.3141%","0.2172%","0.1755%","0.0196%","0.2019%","0.0303%","0.0184%","0.0552%","0.0453%","0.0408%","0.0096%","0.0248%","0.1361%","0.0272%","0.2552%","0.1338%","0.3141%","0.2172%","0.1755%","0.0196%","0.2019%","0.0303%","0.0184%","0.0552%","0.0453%","0.0408%","0.0096%","0.0248%","0.1361%"],["0.6803% ","5.7307% ","4.0035% ","5.3090% ","5.6216% ","2.7747% ","1.1247% ","12.5031%","1.5615% ","-0.1033%","0.1603% ","0.6234% ","3.5402% ","0.2305% ","1.0058% ","4.0892% ","1.7896% ","5.2801% ","3.9277% ","3.9728% ","4.1761% ","1.2228% ","0.1337% ","12.3762%","0.1318% ","0.9894% ","0.8785% ","1.4500% ","1.6014% ","0.1177% ","0.5355% ","5.3974% ","0.8909% ","4.5797% ","3.8501% ","8.5672% ","11.3033%","6.7124% ","2.1844% ","9.5793% ","1.0418% ","0.8818% ","-0.1995%","2.5299% ","1.3860% ","-0.0723%","1.3660% ","7.3497% "],[" 25.0"," 22.5"," 29.9"," 16.9"," 25.9"," 15.8"," 57.4"," 61.9"," 51.5"," -5.6","  2.9"," 13.7"," 86.7"," 23.9"," 40.5"," 30.0"," 65.7"," 20.7"," 29.4"," 12.6"," 19.2","  7.0","  6.8"," 61.3","  4.3"," 53.9"," 15.9"," 32.0"," 39.2"," 12.2"," 21.6"," 39.6"," 32.7"," 17.9"," 28.8"," 27.3"," 52.0"," 38.2","111.6"," 47.4"," 34.4"," 48.0"," -3.6"," 55.8"," 33.9"," -7.5"," 55.0"," 54.0"],["2.4e-01","1.1e-02","9.0e-03","1.1e-01","2.7e-02","1.5e-01","4.9e-02","5.7e-05","3.4e-01","7.9e-01","9.0e-01","4.7e-01","3.7e-02","4.6e-01","2.0e-01","2.5e-02","2.7e-02","6.8e-02","3.8e-02","1.8e-01","1.5e-01","6.0e-01","8.6e-01","1.6e-05","9.0e-01","2.1e-01","4.2e-01","2.1e-01","1.8e-01","7.5e-01","4.4e-01","1.2e-02","2.8e-01","6.7e-02","3.1e-02","1.1e-02","2.2e-02","1.1e-01","2.6e-02","5.4e-03","4.2e-01","2.9e-01","8.1e-01","1.1e-01","1.9e-01","8.5e-01","1.4e-01","2.3e-03"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>trait<\/th>\n      <th>Category<\/th>\n      <th>Prop_SNPs<\/th>\n      <th>Prop_h2<\/th>\n      <th>Enrichment<\/th>\n      <th>Enrichment_p<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"trait","targets":1},{"name":"Category","targets":2},{"name":"Prop_SNPs","targets":3},{"name":"Prop_h2","targets":4},{"name":"Enrichment","targets":5},{"name":"Enrichment_p","targets":6}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>The high standard errors for enrichment estimates are probably due to
low proportion of SNPs present in peaks called by each cluster. To make
the point that lung tissue is more important than spleen to disease, we
can temporarily ignore sub-types but use major immune cell types like T,
B and NK cells to improve confidence interval.</p>
</div>
<div id="test-individual-enrichment-for-major-cell-types-of-each-tissue"
class="section level3">
<h3>Test individual enrichment for major cell types of each tissue</h3>
<p><strong>A barplot for LDSC enrichment results</strong>. Baseline
annotation: baseline v1.2 (53 annotations) recommended for comparing the
enrichment p-values across cell types or tissues. Y-axis:
annotation(%GWAS SNPs within annotation). X-axis: enrichment fold.<br />
Label: enrichment p-value.</p>
<p>Merging cell types to contain more risk variants does help lower the
standard errors. For both tissues, T cells show the strongest enrichment
and contain the highest proportion of risk variants. Across cell types,
lung immune cells show more significant enrichment compare to spleen
immune cells.</p>
<p><img src="figure/u19_h2g_enrichment.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/u19_h2g_enrichment.Rmd/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/u19_h2g_enrichment.Rmd/unnamed-chunk-7-3.png" width="672" style="display: block; margin: auto;" /></p>
<div class="datatables html-widget html-fill-item" id="htmlwidget-9c8a3c9d2f6933833caa" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-9c8a3c9d2f6933833caa">{"x":{"filter":"none","vertical":false,"data":[["9830","9831","9832","9833","9834","9835","9836","9837","9838","9839","9840","9841","9842","9843","9844","9845","9846","9847"],["allergy","allergy","allergy","allergy","allergy","allergy","aoa_ukb","aoa_ukb","aoa_ukb","aoa_ukb","aoa_ukb","aoa_ukb","coa_ukb","coa_ukb","coa_ukb","coa_ukb","coa_ukb","coa_ukb"],["lungs_T_cells","lungs_B_cells","lungs_NK_cells","spleens_T_cells","spleens_B_cells","spleens_NK_cells","lungs_T_cells","lungs_B_cells","lungs_NK_cells","spleens_T_cells","spleens_B_cells","spleens_NK_cells","lungs_T_cells","lungs_B_cells","lungs_NK_cells","spleens_T_cells","spleens_B_cells","spleens_NK_cells"],["0.614%","0.448%","0.282%","0.211%","0.101%","0.049%","0.614%","0.448%","0.282%","0.211%","0.101%","0.049%","0.614%","0.448%","0.282%","0.211%","0.101%","0.049%"],["17.872%","8.407% ","6.129% ","8.179% ","0.758% ","1.421% ","14.730%","6.994% ","6.457% ","7.164% ","2.224% ","1.085% ","24.123%","11.665%","5.196% ","9.509% ","2.160% ","1.873% "],["29.1","18.8","21.7","38.7"," 7.5","29.2","24.0","15.6","22.9","33.9","22.1","22.3","39.3","26.0","18.4","45.0","21.5","38.5"],["4.3e-06","1.9e-02","1.0e-02","1.4e-03","5.7e-01","4.1e-01","9.6e-05","3.5e-02","3.4e-02","6.0e-03","1.4e-01","3.5e-01","3.5e-04","2.2e-03","5.9e-02","1.3e-04","2.2e-01","2.2e-01"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>trait<\/th>\n      <th>Category<\/th>\n      <th>Prop_SNPs<\/th>\n      <th>Prop_h2<\/th>\n      <th>Enrichment<\/th>\n      <th>Enrichment_p<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"trait","targets":1},{"name":"Category","targets":2},{"name":"Prop_SNPs","targets":3},{"name":"Prop_h2","targets":4},{"name":"Enrichment","targets":5},{"name":"Enrichment_p","targets":6}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><strong>A dotplot for LDSC enrichment results</strong></p>
<p>dot size: enrichment fold color: enrichment p-value
<img src="figure/u19_h2g_enrichment.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/ca85982484a362036196c17bf218d91ecaea4872/docs/figure/u19_h2g_enrichment.Rmd/unnamed-chunk-8-1.png" target="_blank">ca85982</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-06-05
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/71a11fc1d56cbc833a77f52979b2db80daabb2cb/docs/figure/u19_h2g_enrichment.Rmd/unnamed-chunk-8-1.png" target="_blank">71a11fc</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-06-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="tissue-specific-analyses-for-each-major-cell-type."
class="section level3">
<h3>Tissue-specific analyses for each major cell type.</h3>
<p>LDSC operates cell-type specific analyses by taking two sets of LD
scores for each test. Here I input one set of LD scores derived from
lung and the other set from spleen as control for each major cell type.
The resulting p-value tests whether the coefficient is greater than
zero, which implies if lung tissue is more significant. To compute
tissue-specific effects, I can obtain peaks specific to lung tissue and
then repeat this analysis.</p>
<p><strong>A barplot for tissue-group analysis</strong> X-axis: -log10
p-values for testing the coefficient
<img src="figure/u19_h2g_enrichment.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/lung_lymph_scMultiomics/blob/ca85982484a362036196c17bf218d91ecaea4872/docs/figure/u19_h2g_enrichment.Rmd/unnamed-chunk-10-1.png" target="_blank">ca85982</a>
</td>
<td>
Jing Gu
</td>
<td>
2024-06-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="jointly-test-multiple-annotations" class="section level3">
<h3>Jointly test multiple annotations</h3>
<p><strong>A barplot for LDSC enrichment results</strong></p>
<p>Baseline annotation: BaselineLD_v2.2 (97 annotations) recommended for
estimating heritability. Y-axis: annotation(%GWAS SNPs within
annotation). X-axis: enrichment fold.<br />
Label: heritability</p>
<p>The chromatin accessible peaks from T cells, B cells, and NK cells
were jointly tested for each tissue. The results show that lung immune
cells explain more disease heritability than spleen immmune cells.
Conditional on other lung immune cells, Treg OCRs have the most
contribution in Allergy and AOA, followed by B cells and NK cells. For
COA, the disease heritability was explained evenly by Treg, CD4+ T and
naive B cells. The CD8+ T cells only contributes to COA heritability.
<img src="figure/u19_h2g_enrichment.Rmd/unnamed-chunk-11-1.png" width="768" style="display: block; margin: auto;" /><img src="figure/u19_h2g_enrichment.Rmd/unnamed-chunk-11-2.png" width="768" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="identify-relevant-cell-types-and-tissues-for-asthma"
class="section level2">
<h2>Identify relevant cell-types and tissues for Asthma</h2>
<p>Based on the list of enhancers ranked by the causal signals from GWAS
and other functional data, I looked for peaks in our dataset that
overlapped with high-confident enhancers. For enhancers overlapped by
multiple peaks, the peak scores were computed by the average of their
significant scores from peak calling.</p>
<p><strong>Heatmap for peak scores</strong></p>
<ul>
<li>X-axis denotes enhancer-gene pair, ranked by either AOA ePIP or COA
ePIP.</li>
<li>Color of heatmap is the log2 value of mean significant scores for
peaks that overlap with enhancer of high PIPs.</li>
</ul>
<div id="aoa-enhancers-ranked-by-epips" class="section level3">
<h3>AOA enhancers ranked by ePIPs</h3>
<p><img src="figure/u19_h2g_enrichment.Rmd/unnamed-chunk-15-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="coa-enhancers-ranked-by-epips" class="section level3">
<h3>COA enhancers ranked by ePIPs</h3>
<p><img src="figure/u19_h2g_enrichment.Rmd/unnamed-chunk-16-1.png" width="768" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] rhdf5_2.42.1                SummarizedExperiment_1.28.0
 [3] Biobase_2.58.0              MatrixGenerics_1.10.0      
 [5] Rcpp_1.0.12                 Matrix_1.6-5               
 [7] matrixStats_1.2.0           stringr_1.5.1              
 [9] plyr_1.8.9                  magrittr_2.0.3             
[11] gtable_0.3.5                gtools_3.9.5               
[13] gridExtra_2.3               ArchR_1.0.2                
[15] rtracklayer_1.58.0          data.table_1.15.4          
[17] GenomicRanges_1.50.2        GenomeInfoDb_1.34.9        
[19] IRanges_2.32.0              S4Vectors_0.36.2           
[21] BiocGenerics_0.44.0         dplyr_1.1.4                
[23] ggplot2_3.4.0               workflowr_1.7.1            

loaded via a namespace (and not attached):
 [1] bitops_1.0-7             fs_1.6.4                 httr_1.4.7              
 [4] rprojroot_2.0.4          tools_4.2.0              bslib_0.7.0             
 [7] utf8_1.2.4               R6_2.5.1                 DT_0.33                 
[10] colorspace_2.1-0         rhdf5filters_1.10.1      withr_3.0.0             
[13] tidyselect_1.2.1         processx_3.8.3           compiler_4.2.0          
[16] git2r_0.33.0             cli_3.6.2                DelayedArray_0.24.0     
[19] labeling_0.4.3           sass_0.4.9               scales_1.3.0            
[22] callr_3.7.3              digest_0.6.35            Rsamtools_2.14.0        
[25] rmarkdown_2.26           XVector_0.38.0           pkgconfig_2.0.3         
[28] htmltools_0.5.8.1        fastmap_1.1.1            highr_0.10              
[31] htmlwidgets_1.6.4        rlang_1.1.3              rstudioapi_0.15.0       
[34] jquerylib_0.1.4          BiocIO_1.8.0             generics_0.1.3          
[37] farver_2.1.1             jsonlite_1.8.8           crosstalk_1.2.1         
[40] BiocParallel_1.32.6      RCurl_1.98-1.14          GenomeInfoDbData_1.2.9  
[43] munsell_0.5.1            Rhdf5lib_1.20.0          fansi_1.0.6             
[46] viridis_0.6.5            lifecycle_1.0.4          stringi_1.7.6           
[49] whisker_0.4.1            yaml_2.3.8               zlibbioc_1.44.0         
[52] parallel_4.2.0           promises_1.3.0           crayon_1.5.2            
[55] lattice_0.22-5           Biostrings_2.66.0        cowplot_1.1.3           
[58] knitr_1.46               ps_1.7.6                 pillar_1.9.0            
[61] rjson_0.2.21             codetools_0.2-19         XML_3.99-0.16.1         
[64] glue_1.7.0               evaluate_0.23            getPass_0.2-2           
[67] vctrs_0.6.5              httpuv_1.6.14            cachem_1.0.8            
[70] xfun_0.43                restfulr_0.0.15          later_1.3.2             
[73] viridisLite_0.4.2        tibble_3.2.1             GenomicAlignments_1.34.1</code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
